model/src/pyrenew/metaclass.py

```diff
@@ -693,14 +693,17 @@ def __init__(
         self.transforms = transforms
         self.validate()
 
-    def sample(self, **kwargs) -> tuple:
+    def sample(self, record=True, **kwargs) -> tuple:
```

Comment:
I'd suggest default `False`. I had thought we had an issue to change `DeterministicVariable` to that as well, but I couldn't find one, so I created it #389